src/generator/pollers.ts

```diff
@@ -53,50 +56,60 @@ type ${poller.name} struct {
 	pt pollingTracker
 }
 
-// Done returns true if the polling operation has terminated either in a success case or failure case,
-// otherwise it will return false
-func (p *${poller.name}) Done() bool {
-	return p.pt.hasTerminated()
-}
-
-// ID ... 
-func (p *${poller.name}) ID() string {
-	return ""
+// Poll returns false if there was an error or polling has reached a terminal state
+func (p *${poller.name}) Poll(ctx context.Context) bool {
+	if done, err := p.done(ctx); err != nil {
+		return false
+	} else {
+		return !done
+	}
 }
 
-func (p *${poller.name}) Poll(ctx context.Context) (*${responseType}, error) {
-	if done, err := p.done(ctx); !done || err != nil {
-		return nil, err
-	}
+// Response returns the final response from the polling operation
+func (p *${poller.name}) Response() (*${responseType}, error) {
 	resp := p.response()
+	if resp == nil {
+		return nil, errors.New("did not find a response on the poller")
+	}
 	result, err := p.client.${poller.operationName}HandleResponse(resp)
 	if err != nil {
 		return nil, err
 	}
 	return result, nil
 }
 
+// ResumeToken generates the string token that can be used with the Resume${pascalCase(poller.name)} method
+// on the client to create a new poller from the data held in the current poller type
+func (p *${poller.name}) ResumeToken() (string, error) {
+	if p.pt.hasTerminated() {
+		return "", errors.New("cannot create a ResumeToken from a poller in a terminal state")
+	}
+	js, err := json.Marshal(p.pt)
+	if err != nil {
+		return "", fmt.Errorf("json.Marshal: %s", err.Error())
```

Comment:
I would return just the `err` here.

Reply:
No, I think it's fine to return just the err here. It was an automatic reaction to wrap it.

Reply:
Also done.